src/compiler/checker.ts

```diff
@@ -9061,45 +9061,49 @@ namespace ts {
         }
 
         function resolveTypeOfAccessors(symbol: Symbol, writing = false) {
+
+            function getInstanceOrType(type: Type, symbol: Symbol) {

```

Comment:
Two nits:

1. `getInstantiatedType` or `instantiateIfNeeded` might be a better name; “instance” is an overloaded word even though it’s the root of “instantiate.”
2. The loose convention for these inner functions seems to be putting them at the bottom, after the outer function’s `return`.

Reply:
`instantiateTypeIfNeeded` was my first choice but decided to change it because I was unsure if people like it. I submitted that on the latest revision.

Changed the inner function to after the `return`